src/testnet/run_loop.rs

```diff
@@ -14,7 +14,7 @@ pub struct RunLoop {
     nodes: Vec<Node>,
     new_burnchain_state_callback: Option<fn(u8, &BurnchainState)>,
     new_tenure_callback: Option<fn(u8, &LeaderTenure)>,
-    new_chain_state_callback: Option<fn(u8, &StacksChainState)>,
+    new_chain_state_callback: Option<fn(u8, &mut StacksChainState, &BlockHeaderHash)>,
```

Comment:
Why do we have to make the arg `StacksChainState` mutable?

Reply:
To query the chain state information -- since that information is in the MARF, reads require a mutable reference.